pxr/imaging/hdSt/materialXShaderRegistry.cpp

```diff
@@ -0,0 +1,542 @@
+﻿//
+// Copyright 2020 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include "pxr/base/tf/envSetting.h"
+#include "pxr/base/js/json.h"
+
+#include "pxr/imaging/hdSt/materialXShaderRegistry.h"
+#include "pxr/imaging/hdSt/materialXShaderGen.h"
+
+#include "pxr/base/gf/vec2f.h"
+#include "pxr/base/gf/vec3f.h"
+#include "pxr/base/gf/vec4f.h"
+
+#include <MaterialXGenShader/Shader.h>
+
+#include <fstream>
+
+namespace mx = MaterialX;
+namespace fs = std::filesystem;
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+TF_DEFINE_ENV_SETTING(HDST_MTLX_CODEGEN_CACHE_DIR_PATH, "",
+    "Path to the directory of the persistent MaterialX codegen cache");
```

Comment:
Environment variable which can be used to set the path to the directory where the cached files are written. It serves as the default value for an `HdSt` render setting.